inc/item_devices.class.php

```diff
@@ -1142,16 +1162,15 @@ static function cleanItemDeviceDBOnItemDelete($itemtype, $items_id, $unaffect) {
          $link = getItemForItemtype($link_type);
          if ($link) {
             if ($unaffect) {
-               $query = "SELECT `id`
-                         FROM `".$link->getTable()."`
-                         WHERE `itemtype` = '$itemtype'
-                               AND `items_id` = '$items_id'";
-               $input = ['items_id' => 0,
-                              'itemtype' => ''];
-               foreach ($DB->request($query) as $data) {
-                  $input['id'] = $data['id'];
-                  $link->update($input);
-               }
+               $DB->update(
+                  $link->getTable(), [
+                     'items_id'  => 0,
+                     'itemtype'  => ''
+                  ], [
+                     'items_id'  => $items_id,
```

Comment:
i 'm not sure we can relace this update, as you miss the `input['id']` part

Reply:
hum, maybe not, it's difficult to say it's exactly the same

Reply:
Non, `$input['id']` is required for `CommonDBTM::update()` but not for `$DB::update()`  ;)